pkg/subctl/cmd/gather.go

```diff
@@ -129,22 +124,15 @@ func gatherDataByCluster(restConfig restConfig) {
 	}
 
 	info.DynClient, info.ClientSet, err = getClients(restConfig.config)
-	exitOnError("Error getting client %s", err)
-
-	resourceSubmariners := schema.GroupVersionResource{
-		Group:    submarinerOp.SchemeGroupVersion.Group,
-		Version:  submarinerOp.SchemeGroupVersion.Version,
-		Resource: "submariners",
+	if err != nil {
+		fmt.Printf("Error getting client: %s\n", err)
+		return
 	}
 
-	resourceSubm, err := info.DynClient.Resource(resourceSubmariners).Namespace(submarinerNamespace).Get("submariner", metav1.GetOptions{})
-	if err == nil {
-		var submariner submarinerOp.Submariner
-		err = runtime.DefaultUnstructuredConverter.FromUnstructured(resourceSubm.UnstructuredContent(), &submariner)
-		if err != nil {
-			exitOnError(fmt.Sprintf("Failed to get submariner resource %q", err), err)
-		}
-		submarinerCR = &submariner
+	info.Submariner, err = getSubmarinerResourceWithError(restConfig.config)
+	if err != nil && !apierrors.IsNotFound(err) {
```

Comment:
Submariner CRD is not installed on broker? If CRD isn't even present, we will not get IsNotFound but a different error. In case of ServiceDiscovery only deployments, it might be present. Either ways, `IsNotFound()` by itself is not reliable enough.

Reply:
Yes, to install the operator we install both CRDs, then you create the CRs or not..